doc/pluginsignals.c

```diff
@@ -92,6 +92,17 @@ signal void (*document_reload)(GObject *obj, GeanyDocument *doc, gpointer user_d
  */
 signal void (*document_before_save)(GObject *obj, GeanyDocument *doc, gpointer user_data);
 
+/** Sent before save as is performed with the original document.
+ *
+ * @param obj a GeanyObject instance, should be ignored.
+ * @param doc the original document. The document with the new file name is still
+ *        reported by the "document-save" signal sent afterwards.
+ * @param user_data user data.
+ *
+ * @since TODO
+ */
+signal void (*document_before_save_as)(GObject *obj, GeanyDocument *doc, gpointer user_data);
```

Comment:
Perhaps add a `boolean` return value to potentially allow the plugin to block saving the file?

Reply:
Hmm, I'm not sure if plugins should be allowed to do that and what exactly would this be good for. Do you have some example?

Reply:
No, the plugin should not block a user saving a file, under any circumstances.